### PR TITLE
Catch and report errors connecting to Autograph in checks

### LINKED ISSUE #1607: Heartbeat should catch and properly report inability to connect to Autograph
https://sentry.prod.mozaws.net/operations/normandy-stage/issues/4903226/

```
gaierror: [Errno -2] Name or service not known
  File "urllib3/connection.py", line 159, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File "urllib3/util/connection.py", line 57, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "python3.6/socket.py", line 745, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):

NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x7f1d775b20b8>: Failed to establish a new connection: [Errno -2] Name or service not known
  File "urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "urllib3/connectionpool.py", line 343, in _make_request
    self._validate_conn(conn)
  File "urllib3/connectionpool.py", line 839, in _validate_conn
    conn.connect()
  File "urllib3/connection.py", line 301, in connect
    conn = self._new_conn()
  File "urllib3/connection.py", line 168, in _new_conn
    self, "Failed to establish a new connection: %s" % e)

MaxRetryError: HTTPSConnectionPool(host='content-signature.stage.mozaws.net', port=443): Max retries exceeded with url: /chains/normandy.content-signature.mozilla.org-20191015.stage.chain (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f1d775b20b8>: Failed to establish a new connection: [Errno -2] Name or service not known',))
  File "requests/adapters.py", line 449, in send
    timeout=timeout
  File "urllib3/connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "urllib3/util/retry.py", line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))

ConnectionError: HTTPSConnectionPool(host='content-signature.stage.mozaws.net', port=443): Max retries exceeded with url: /chains/normandy.content-signature.mozilla.org-20191015.stage.chain (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f1d775b20b8>: Failed to establish a new connection: [Errno -2] Name or service not known',))
(8 additional frame(s) were not displayed)
...
  File "dockerflow/django/middleware.py", line 45, in process_request
    return view(request)
  File "dockerflow/django/views.py", line 57, in heartbeat
    detail = heartbeat_check_detail(check)
  File "dockerflow/django/views.py", line 79, in heartbeat_check_detail
    errors = check(app_configs=None)

ConnectionError: HTTPSConnectionPool(host='content-signature.stage.mozaws.net', port=443): Max retries exceeded with url: /chains/normandy.content-signature.mozilla.org-20191015.stage.chain (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f1d775b20b8>: Failed to establish a new connection: [Errno -2] Name or service not known',))
```

### PR DESCRIPTION
**Pull request by mythmon**
Fixes #1607

### CODE REVIEW

File: normandy/recipes/checks.py [hunk 1]
```diff
@@ -102,20 +105,32 @@ def signatures_use_good_certificates(app_configs, **kwargs):
...
-                bad_objects = list(matching_recipes) + list(matching_actions)
-
-                object_names = ", ".join(str(o) for o in bad_objects)
+                bad_object_names = get_matching_object_names(url)
+                msg = (
+                    f"{len(bad_object_names)} objects are signed with a bad cert: "
+                    f"{bad_object_names}. {exc.detail}. Certificate url is {url}. "
+                )
+                errors.append(Error(msg, id=ERROR_BAD_SIGNING_CERTIFICATE))
+            except requests.ConnectionError as exc:
```
Comment (peterbe) [inline normandy/recipes/checks.py]: Is it only `ConnectionError`s we've seen in Sentry? We could consider using `RequestException`.

Comment (peterbe) [reply normandy/recipes/checks.py]: That would include things like `Timeout`.

Comment (peterbe) [reply normandy/recipes/checks.py]: Actually, there have been at least [2 `HTTPError`s](https://sentry.prod.mozaws.net/operations/normandy-stage/issues/4951178/?query=is:unresolved) that would be missed by `ConnectionError`. 

Seems the simplest and safest is to just use `except requests.RequestException as exc:`

**Comment by mythmon** [comment]
> I'm guessing something built into python-dockerflow or something already takes care of dumping these things to logging. Have you confirmed that?

It does not, and python-dockerflow doesn't expose hooks we could use to add them. We'll have to add them upstream. I think we should also work with @sciurus to figure out what this logging should be anyways.

**Comment by sciurus** [comment]
Sorry, I missed that I was mentioned here earlier. How about if the heartbeat endpoint was requested and the status is not okay, log the failing checks.  Something like

```
{
  "Timestamp":1543536058688097536,
  "Type":"heartbeat",
  "Logger":"normandy",
  "Hostname":"ip-blah.us-west-2.compute.internal",
  "EnvVersion":"2.0",
  "Severity":3,
  "Pid":99,
  "Fields":{
    "failing_check_name": "failing_check_message",
    "another_failing_check_name": "another_failing_check_message"
  }
}
```

**Comment by sciurus** [comment]
Actually after asking around it turns out this logging isn't necessary.

I didn't think Pingdom recorded responses because I couldn't find them in the UI. Turns out that's because it only saves and displays them if the check has failed. Which is exactly what we want. 😁

**Comment by peterbe** [comment]
It works!!
Short version...

**Exhibit A**
screen shot 2018-11-30 at 3 15 51 pm

**Exhibit B**
screen shot 2018-11-30 at 3 16 24 pm

I'll explain more what I had to do to test this...

**Comment by peterbe** [comment]
Ignore these last comments above by me. I was tab-confused and commented on the wrong PR. Will delete them.

**Comment by mythmon** [comment]
Sounds like we're all good here then.

bors r= @peterbe, @sciurus

**Pull request merged**